src/fix/fixField.h

```diff
@@ -0,0 +1,371 @@
+
+
+namespace neueda
+{
+
+using namespace std;
+
+class fixField;
+
+typedef bool (fixField::*getAsFunc) (const char*, const int64_t, cdr&);
+typedef codecState (fixField::*setAsFunc) (const int64_t,
+                                           const cdrItem&,
+                                           const size_t len,
+                                           char* p,
+                                           size_t& used);
+
+class fixField
+{
+public:
+    fixField () :
+        mRequired (false),
+        mTag (0)
+    {
+        string t ("STRING");
+        setType (t);
+    }
+
+    bool getAsString (const char* val, const int64_t tag, cdr& d)
+    {
+        string value (val);
+        utils_trimWhiteSpace (value);
+
+        d.setString (tag, value);
+
+        return true;
+    }
+
+    static codecState writeStringVal (const int64_t tag,
+                                      const string& val,
+                                      const size_t len,
+                                      char* p,
+                                      size_t& used)
+    {
+        char* tmp = p + used;
+        tmp += snprintf (p + used, len - used, "%lld=%s\001", tag, val.c_str ());
+        used += tmp - (p + used);
+
+        if (used > len)
+            return GW_CODEC_SHORT;
+
+        return GW_CODEC_SUCCESS;
+    }
+
+    static codecState setAsString (const int64_t tag,
+                                   const cdrItem& i,
+                                   const size_t len,
+                                   char* p,
+                                   size_t& used)
+    {
+        string val;
+        if (!i.asString(val))
+            return GW_CODEC_ERROR;
+
+        return writeStringVal (tag, val, len, p, used);
+    }
+
+    bool get (const char* val, cdr& d)
+    {
+        return ((this)->*mGetter) (val, mTag, d);
+    }
+
+    codecState set (const int64_t tag,
+                    const cdrItem& i,
+                    const size_t len,
+                    char* p,
+                    size_t& used)
+    {
+        if (mSetter != NULL)
+            return ((this)->*mSetter) (tag, i, len, p, used);
+        else
+            return fixField::setAsString (tag, i, len, p, used);
+    }
+
+    bool required () const
+    {
+        return mRequired;
+    }
+
+    void setRequired (bool req)
+    {
+        mRequired = req;
+    }
+
+    int64_t tag () const
+    {
+        return mTag;
+    }
+
+    void setTag (int64_t tag)
+    {
+        mTag = tag;
+    }
+
+    string getName () const
+    {
+        return mName;
+    }
+
+    void setName (const string& name)
+    {
+        mName = name;
+    }
+
+    void setType (const string& type)
+    {
+        mType = type;
+
+        // TODO DATA: Raw data with no format or content restrictions
+        if (type == "INT" || type == "LENGTH" || type == "DAYOFMONTH")
+        {
+            mGetter = &fixField::getAsInteger;
+        }
+        else if (type == "UTCTIMESTAMP")
```

Comment:
Instead of using strings for type you could a enum means you could then collapse case statements rather than having all these if else

Reply:
I don't see the value - The XML doc is parsed and those are the values contained in the XML doc. I would have to compute the enums when i read the doc so this code would just move. Its only used on initialisation anyways

C